qiskit/visualization/matplotlib.py

```diff
@@ -276,16 +226,72 @@ def _registers(self, creg, qreg):
 
     @property
     def ast(self):
+        """AST getter"""
         return self._ast
 
+    def _load_style(self, style):
+        current_style = DefaultStyle().style
+        style_name = 'default'
+        config = user_config.get_config()
+        if style is False:
+            style_name = 'bw'
+        elif isinstance(style, dict) and 'name' in style.keys():
+            style_name = style['name']
+        elif isinstance(style, str):
+            style_name = style
+        elif config:
+            style_name = config.get('circuit_mpl_style', 'default')
+        if style_name[-5:] == '.json':
```

Comment:
if `style_name` is shorter than 5, fails. Better 
```suggestion
        if style_name.endswith(.json'):
```

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/5223/commits/bfb8be30672ba6d81187f57c58a4079026e0bb0e